File: code/r/generate_data/mean_shift.R
get_dense_shift_normal_mean <- function(delta, n, p, prop = 0.2, tau = 0.5){
  d <- floor(p * prop)
  shift <- delta / sqrt(d)
  t0 <- floor(n * tau)
  s1 <- matrix(0, t0, p)
  s2 <- matrix(0, (n - t0), p)
  for(j in 1:p)
    s1[, j] <- rnorm(t0)
  for(j in 1:p){
    if(j <= d)
      s2[, j] <- rnorm(n-t0, shift)
    else
      s2[, j] <- rnorm(n-t0)
  }
  rbind(s1, s2)
}

get_sparse_shift_normal_mean <- function(delta, n, p, prop = 0.01, 
                                        tau = 0.5){
  d <- floor(p * prop)
  shift <- delta / sqrt(d)
  t0 <- floor(n * tau)
  s1 <- matrix(0, t0, p)
  s2 <- matrix(0, (n - t0), p)
  for(j in 1:p)
    s1[, j] <- rnorm(t0)
  for(j in 1:p){
    if(j <= d)
      s2[, j] <- rnorm(n-t0, shift)
    else
      s2[, j] <- rnorm(n-t0)
  }
  rbind(s1, s2)
}

File: code/py/generate_data/mean_shift.py
import numpy as np


def get_dense_shift_normal_mean(delta, n, p, prop=0.2, tau=0.5):
    d = int(np.floor(p * prop))
    shift = delta / np.sqrt(d)
    t0 = int(np.floor(n * tau))
    mu1 = np.zeros(p)
    mu2 = np.zeros(p)
    mu2[np.arange(d)] = shift
    sigma = np.diag(np.ones(p))
    s1 = np.random.multivariate_normal(mu1, sigma, t0)
    s2 = np.random.multivariate_normal(mu2, sigma, n - t0)
    sample = np.concatenate((s1, s2), axis=0)

    return sample


def get_sparse_shift_normal_mean(delta, n, p, prop=0.01, tau=0.5):
    d = int(np.floor(p * prop))
    shift = delta / np.sqrt(d)
    t0 = int(np.floor(n * tau))
    mu1 = np.zeros(p)
    mu2 = np.zeros(p)
    mu2[np.arange(d)] = shift
    sigma = np.diag(np.ones(p))
    s1 = np.random.multivariate_normal(mu1, sigma, t0)
    s2 = np.random.multivariate_normal(mu2, sigma, n - t0)
    sample = np.concatenate((s1, s2), axis=0)

    return sample
